src/Umbraco.Infrastructure/Mail/Interfaces/IEmailSenderClient.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using MimeKit;
+
+namespace Umbraco.Cms.Infrastructure.Mail.Interfaces
+{
+    /// <summary>
+    /// Client for sending an email from a MimeMessage
+    /// </summary>
+    public interface IEmailSenderClient
+    {
+        /// <summary>
+        /// Sends the email message
+        /// </summary>
+        /// <param name="message"></param>
+        /// <returns></returns>
+        public Task SendAsync(MimeMessage? message);
```

Comment:
Fundamentally I do not like the fact that we expose a third party class in this public interface.

I would prefer it to be our own model, and I'm not sure why it is nullable?

Reply:
@bergmania nullability must have been a brainfart.

We can expose your model for the interface, but then I think we should make your extension to turn it into a mimemessage public, as the whole exercise is to make it easier to extend the emailclient without having to re-implement too much of your logic.

I've updated the pull-request with a non-nullable "EmailMessage" model and moved the MimeMessage part into the implementation as well as exposed its extention from internal to public

Reply:
But the issue with exposing third party classes is that we cannot replace this underlying framework without braking our contracts/interfaces. - Ideally we should be able to implement our contracts with all mail client libraries.

Reply:
I think moving the interface to use your model honors the ability to implement contracts with all client libraries.

I understand the logic behind the extension method in principle, but also just see how it's just going to lead to us on the other side of the fence having to write the exact same code anyway.

But if that's a hard no, I'll update the pull request and steal your class for my implementation and package.

Reply:
Noted. I've updated the extension class to be internal again.